megamek/src/megamek/client/bot/common/FacingDiffCalculator.java

```diff
@@ -0,0 +1,193 @@
+          @Nullable final Coords enemyMedianPosition, @Nullable final Coords closestEnemyPosition) {
+        Coords toFace = selectOneCoordsToFace(path, secondaryTargetPosition, enemyMedianPosition, closestEnemyPosition);
+        int desiredFacing = getDesiredFacing(unit, path, toFace);
+        int currentFacing = path.getFinalFacing();
+        return getFacingDiff(currentFacing, desiredFacing);
+    }
+
+    /**
+     * Calculates the optimal facing direction for a unit toward a target coordinate.
+     * <p>
+     * This method accounts for armor distribution biases that may make it advantageous to face slightly off-center from
+     * the target.
+     * </p>
+     *
+     * @param unit   The entity making the movement
+     * @param path   The movement path being evaluated
+     * @param toFace The target coordinates to face toward
+     *
+     * @return The desired facing direction (0-5)
+     */
+    private int getDesiredFacing(Entity unit, MovePath path, Coords toFace) {
+        int desiredFacing = path.getFinalCoords().direction(toFace);
+
+        // -1 is bias towards facing left, 1 is bias towards facing right
+        desiredFacing += getBiasTowardsFacing(unit);
+        if (desiredFacing < 0) {
+            desiredFacing += 6;
+        }
+        desiredFacing %= 6;
+        return desiredFacing;
+    }
+
+    /**
+     * Selects which coordinates the unit should face toward.
+     * <p>
+     * Priority is given to:
+     * <ol>
+     *   <li>Adjacent enemies (if the path ends adjacent to the closest enemy)</li>
+     *   <li>Median enemy position (if enemies are present but not adjacent)</li>
+     *   <li>Secondary target position (if no enemies are present)</li>
+     * </ol>
+     *
+     * @param path                 The movement path being evaluated
+     * @param secondaryPosition    Fallback position if no enemies are present
+     * @param enemyMedianPosition  The median position of nearby enemies
+     * @param closestEnemyPosition The position of the closest enemy
+     *
+     * @return The coordinates the unit should face toward
+     */
+    private static Coords selectOneCoordsToFace(MovePath path, Coords secondaryPosition, Coords enemyMedianPosition,
+          Coords closestEnemyPosition) {
+        Coords toFace;
+        if ((closestEnemyPosition != null) && closestEnemyPosition.distance(path.getFinalCoords()) == 1) {
+            // The closestEnemyPosition is the position of the closest enemy, if we are moving in direction of it and
+            // landing beside it (reason why dist == 1), we want to face it.
+            toFace = closestEnemyPosition;
+        } else if (enemyMedianPosition != null) {
+            // Otherwise, we face the median position of the closest units
+            toFace = enemyMedianPosition;
+        } else {
+            // If we don't have a target, we face the center of the board
```

Comment:
Is the secondary position guaranteed to be the center of the board, or should we just say, 
```
// If we don't have a target, face the provided secondary position
```
?

Reply:
```java

    /**
     * Selects which coordinates the unit should face toward.
     * <p>
     * Priority is given to:
     * <ol>
     *   <li>Adjacent enemies (if the path ends adjacent to the closest enemy)</li>
     *   <li>Median enemy position (if enemies are present but not adjacent)</li>
     *   <li>Secondary target position (if no enemies are present)</li>
     * </ol>
     *
     * @param path                 The movement path being evaluated
     * @param secondaryPosition    Fallback position if no enemies are present
     * @param enemyMedianPosition  The median position of nearby enemies
     * @param closestEnemyPosition The position of the closest enemy
     *
     * @return The coordinates the unit should face toward
     */
  ```